Source/ResponseSerialization.swift

```diff
@@ -253,18 +253,19 @@ extension DownloadRequest {
 extension Request {
     /// Returns a result data type that contains the response data as-is.
     ///
+    /// - parameter request:  The request performed.
     /// - parameter response: The response from the server.
     /// - parameter data:     The data returned from the server.
     /// - parameter error:    The error already encountered if it exists.
     ///
     /// - returns: The result data type.
-    public static func serializeResponseData(response: HTTPURLResponse?, data: Data?, error: Error?) -> Result<Data> {
+    public static func serializeResponseData(request: URLRequest?, response: HTTPURLResponse?, data: Data?, error: Error?) -> Result<Data> {
```

Comment:
Unfortunately I don't see any other way to check the request type without passing the `URLRequest` into the serializer.